src/clj/ataru/virkailija/virkailija_routes.clj

```diff
@@ -200,54 +200,46 @@
-                                                           {}))]
-                             (response/ok {:applications (for [application applications
-                                                               :let [person      (get persons (:person-oid application))
-                                                                     yksiloity   (or (-> person :yksiloity)
-                                                                                     (-> person :yksiloityVTJ))
-                                                                     person-info (if yksiloity
-                                                                                   {:preferred-name (:kutsumanimi person)
-                                                                                    :last-name      (:sukunimi person)}
-                                                                                   (select-keys application [:preferred-name :last-name]))]]
-                                                           (-> application
-                                                               (assoc :person person-info)
-                                                               (dissoc :person-oid :preferred-name :last-name)))})))
+                   (api/GET "/list" {session :session}
+                     :query-params [{formKey :- s/Str nil}
+                                    {hakukohdeOid :- s/Str nil}
+                                    {hakuOid :- s/Str nil}
+                                    {ssn :- s/Str nil}
+                                    {dob :- s/Str nil}
+                                    {email :- s/Str nil}
+                                    {name :- s/Str nil}
+                                    {personOid :- s/Str nil}
+                                    {applicationOid :- s/Str nil}]
+                     :summary "Return applications header-level info for form"
+                     :return {:applications [ataru-schema/ApplicationInfo]}
+                     (let [[query-key query-value] (cond
+                                                     (some? formKey) [:form formKey]
+                                                     (some? hakukohdeOid) [:hakukohde-oid hakukohdeOid]
+                                                     (some? hakuOid) [:haku-oid hakuOid]
+                                                     (some? ssn) [:ssn ssn]
+                                                     (some? dob) [:dob (when (dob/dob? dob) dob)]
+                                                     (some? email) [:email email]
+                                                     (some? name) [:name name]
+                                                     (some? personOid) [:person-oid personOid]
+                                                     (some? applicationOid) [:application-oid applicationOid])
```

Comment:
Catch all nil and return bad request?

Reply:
Also could be queryType as query param and use schema enum to match to any of the supported query types?

Reply:
Don't really feel like reworking the query parameters :) Will check for nils though.